chia/data_layer/data_store.py

```diff
@@ -1443,6 +1446,64 @@ async def remove_subscriptions(self, tree_id: bytes32, urls: List[str]) -> None:
                     },
                 )
 
+    async def delete_store_data(self, tree_id: bytes32) -> None:
+        async with self.db_wrapper.writer() as reader:
+            cursor = await reader.execute(
+                """
+                WITH all_nodes AS (
+                    SELECT a.hash, n.left, n.right
+                    FROM ancestors AS a
+                    JOIN node AS n ON a.hash = n.hash
+                    WHERE a.tree_id = ?
+                )
+
+                SELECT hash, left, right
+                FROM all_nodes
+                WHERE hash NOT IN (SELECT hash FROM ancestors WHERE tree_id != ?)
+                """,
+                (
+                    tree_id,
+                    tree_id,
+                ),
```

Comment:
Note the use of `:tree_id` and `{"tree_id": tree_id}` so we are using names to map things.  The same could be applied below for consistency.

```suggestion
                """
                WITH all_nodes AS (
                    SELECT a.hash, n.left, n.right
                    FROM ancestors AS a
                    JOIN node AS n ON a.hash = n.hash
                    WHERE a.tree_id = :tree_id
                )

                SELECT hash, left, right
                FROM all_nodes
                WHERE hash NOT IN (SELECT hash FROM ancestors WHERE tree_id != :tree_id)
                """,
                {"tree_id": tree_id},
```